src/Controller/CaseEventController.php

```diff
@@ -0,0 +1,172 @@
+            $choices[$party->getName().', '.$type] = $party;
+        }
+
+        // Setup form and handle it.
+        $form = $this->createForm(CaseEventNewType::class, null, ['choices' => $choices]);
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $data = $form->getData();
+            $caseEvent = new CaseEvent();
+
+            if (is_array($data['senders'])) {
+                $this->createCaseEventRelations($caseEvent, $manager, CaseEventPartyRelation::TYPE_SENDER, $data['senders']);
+            }
+
+            if (is_array($data['recipients'])) {
+                $this->createCaseEventRelations($caseEvent, $manager, CaseEventPartyRelation::TYPE_RECIPIENT, $data['recipients']);
+            }
+
+            $caseEvent->setSubject($data['subject']);
+            $caseEvent->setNoteContent($data['noteContent']);
+            $caseEvent->setCategory(CaseEvent::CATEGORY_NOTE);
+            $caseEvent->setCaseEntity($case);
+            $caseEvent->setCreatedBy($this->getUser());
+            $caseEvent->setReceivedAt(new DateTime('now'));
+
+            $manager->persist($caseEvent);
+            $manager->flush();
+
+            $this->addFlash('success', new TranslatableMessage('Case event created', [], 'case_event'));
+
+            return $this->redirectToRoute('case_event_index', [
+                'id' => $case->getId()->__toString(),
+            ]);
+        }
+
+        return $this->render('case/event/create.html.twig', [
+            'case' => $case,
+            'form' => $form->createView(),
+        ]);
+    }
+
+    #[Route('/{caseEvent}/edit', name: 'case_event_edit', methods: ['GET', 'POST'])]
+    public function edit(CaseEntity $case, CaseEvent $caseEvent, EntityManagerInterface $manager, Request $request): Response
+    {
+        // Setup form and handle it.
+        $form = $this->createForm(CaseEventEditType::class, $caseEvent);
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $manager->flush();
+
+            $this->addFlash('success', new TranslatableMessage('Case event updated', [], 'case_event'));
+
+            return $this->redirectToRoute('case_event_show', [
+                'id' => $case->getId()->__toString(),
+                'caseEvent' => $caseEvent->getId()->__toString(),
+            ]);
+        }
+
+        return $this->render('case/event/edit.html.twig', [
+            'case' => $case,
+            'form' => $form->createView(),
+        ]);
+    }
+
+    private function createCaseEventRelations(CaseEvent $caseEvent, EntityManagerInterface $manager, string $type, array $parties)
+    {
+        foreach ($parties as $party) {
+            $caseEventPartyRelation = new CaseEventPartyRelation();
+            $caseEventPartyRelation->setParty($party);
+            $caseEventPartyRelation->setCaseEvent($caseEvent);
+            $caseEventPartyRelation->setType($type);
```

Comment:
Chain these setter calls. Maybe even rewrite to

```php
$caseEventPartyRelation = (new CaseEventPartyRelation())
  ->setParty($party)
  ->set…;
```

Reply:
Moved to `CaseEventHelper` and chained.